### PR TITLE
add hikari.locales.Locale

### PR DESCRIPTION
**Pull request by Le0Developer**
### Summary

Add `hikari.locales.Locale` enum. (possibly breaking change?)

Not sure where else to put it, tell me if there's a better place for it. Open for feedback :)

### Checklist

- [x] I have run `nox` and all the pipelines have passed.
- [x] I have made unittests according to the code I have added/modified/deleted.

### CODE REVIEW

File: tests/hikari/impl/test_entity_factory.py [hunk 1]
```diff
@@ -25,6 +25,7 @@
 import mock
 import pytest
 
+from hikari import Locale
```
Comment (davfsa) [inline tests/hikari/impl/test_entity_factory.py:28]:
```suggestion
from hikari import locale
```

And then access as such

File: hikari/locales.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    """Language is Chinese, China (zh-CN)"""
+
+    JA = "ja"
+    """Language is Japanese (ja)"""
+
+    ZH_TW = "zh-TW"
+    """Language is Chinese, Taiwan (zh-TW)"""
+
+    KO = "ko"
+    """Language is Korean (ko)"""
```
Comment (davfsa) [inline hikari/locales.py:127]: I dont think there is a need for the `Language is` prefix. Make it all a bit confusing
```suggestion
    DA = "da"
    """Danish"""

    DE = "de"
    """German"""

    EN_GB = "en-GB"
    """English, UK"""

    EN_US = "en-US"
    """English, US"""

    ES_ES = "es-ES"
    """Spanish, Spain"""

    FR = "fr"
    """French"""

    HR = "hr"
    """Croatian"""

    IT = "it"
    """Italian"""

    LT = "lt"
    """Lithuanian"""

    HU = "hu"
    """Hungarian"""

    NL = "nl"
    """Dutch"""

    NO = "no"
    """Norwegian"""

    OL = "pl"
    """Polish"""

    PT_BR = "pt-BR"
    """Portuguese, Bralizian"""

    RO = "ro"
    """Romian"""

    FI = "fi"
    """Finnish"""

    SV_SE = "sv-SE"
    """Swedish, Sweden"""

    VI = "vi"
    """Vietnamese"""

    TR = "tr"
    """Turkish"""

    CS = "cs"
    """Czech"""

    EL = "el"
    """Greek"""

    BG = "bg"
    """Bulgarian"""

    RU = "ru"
    """Russian"""

    UK = "uk"
    """Ukrainian"""

    HI = "hi"
    """Hindi"""

    TH = "th"
    """Thai"""

    ZH_CN = "zh-CN"
    """Chinese, China"""

    JA = "ja"
    """Japanese"""

    ZH_TW = "zh-TW"
    """Chinese, Taiwan"""

    KO = "ko"
    """Korean"""
```

File: hikari/locales.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""Enum of Discord locales."""
+from __future__ import annotations
+
+__all__ = [
+    "Locale",
+]
```
Comment (davfsa) [inline hikari/locales.py:28]:
```suggestion
__all__ = ["Locale"]
```

File: hikari/__init__.py [hunk 4]
```diff
@@ -103,6 +103,7 @@
 from hikari.interactions.component_interactions import *
 from hikari.invites import *
 from hikari.iterators import *
+from hikari.locales import Locale
```
Comment (davfsa) [inline hikari/__init__.py:106]:
```suggestion
from hikari.locales import *
```

File: hikari/locales.py [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""Enum of Discord locales."""
```
Comment (davfsa) [inline hikari/locales.py:23]:
```suggestion
"""Enum of Discord accepted locale."""

```

File: hikari/locales.py (hunk 2 above)
Comment (davfsa) [inline hikari/locales.py:127]: I dont think there is a need for the `Language is` prefix. Make it all a bit confusing
```suggestion
    DA = "da"
    """Danish"""

    DE = "de"
    """German"""

    EN_GB = "en-GB"
    """English, UK"""

    EN_US = "en-US"
    """English, US"""

    ES_ES = "es-ES"
    """Spanish, Spain"""

    FR = "fr"
    """French"""

    HR = "hr"
    """Croatian"""

    IT = "it"
    """Italian"""

    LT = "lt"
    """Lithuanian"""

    HU = "hu"
    """Hungarian"""

    NL = "nl"
    """Dutch"""

    NO = "no"
    """Norwegian"""

    OL = "pl"
    """Polish"""

    PT_BR = "pt-BR"
    """Portuguese, Bralizian"""

    RO = "ro"
    """Romian"""

    FI = "fi"
    """Finnish"""

    SV_SE = "sv-SE"
    """Swedish, Sweden"""

    VI = "vi"
    """Vietnamese"""

    TR = "tr"
    """Turkish"""

    CS = "cs"
    """Czech"""

    EL = "el"
    """Greek"""

    BG = "bg"
    """Bulgarian"""

    RU = "ru"
    """Russian"""

    UK = "uk"
    """Ukrainian"""

    HI = "hi"
    """Hindi"""

    TH = "th"
    """Thai"""

    ZH_CN = "zh-CN"
    """Chinese, China"""

    JA = "ja"
    """Japanese"""

    ZH_TW = "zh-TW"
    """Chinese, Taiwan"""

    KO = "ko"
    """Korean"""
```

Comment (davfsa) [inline hikari/locales.py:127]: I dont think there is a need for the `Language is` prefix. Make it all a bit confusing
```suggestion
    DA = "da"
    """Danish"""

    DE = "de"
    """German"""

    EN_GB = "en-GB"
    """English, UK"""

    EN_US = "en-US"
    """English, US"""

    ES_ES = "es-ES"
    """Spanish, Spain"""

    FR = "fr"
    """French"""

    HR = "hr"
    """Croatian"""

    IT = "it"
    """Italian"""

    LT = "lt"
    """Lithuanian"""

    HU = "hu"
    """Hungarian"""

    NL = "nl"
    """Dutch"""

    NO = "no"
    """Norwegian"""

    OL = "pl"
    """Polish"""

    PT_BR = "pt-BR"
    """Portuguese, Bralizian"""

    RO = "ro"
    """Romian"""

    FI = "fi"
    """Finnish"""

    SV_SE = "sv-SE"
    """Swedish, Sweden"""

    VI = "vi"
    """Vietnamese"""

    TR = "tr"
    """Turkish"""

    CS = "cs"
    """Czech"""

    EL = "el"
    """Greek"""

    BG = "bg"
    """Bulgarian"""

    RU = "ru"
    """Russian"""

    UK = "uk"
    """Ukrainian"""

    HI = "hi"
    """Hindi"""

    TH = "th"
    """Thai"""

    ZH_CN = "zh-CN"
    """Chinese, China"""

    JA = "ja"
    """Japanese"""

    ZH_TW = "zh-TW"
    """Chinese, Taiwan"""

    KO = "ko"
    """Korean"""
```

Comment (davfsa) [inline hikari/locales.py:127]: I dont think there is a need for the `Language is` prefix. Make it all a bit confusing
```suggestion
    DA = "da"
    """Danish"""

    DE = "de"
    """German"""

    EN_GB = "en-GB"
    """English, UK"""

    EN_US = "en-US"
    """English, US"""

    ES_ES = "es-ES"
    """Spanish, Spain"""

    FR = "fr"
    """French"""

    HR = "hr"
    """Croatian"""

    IT = "it"
    """Italian"""

    LT = "lt"
    """Lithuanian"""

    HU = "hu"
    """Hungarian"""

    NL = "nl"
    """Dutch"""

    NO = "no"
    """Norwegian"""

    OL = "pl"
    """Polish"""

    PT_BR = "pt-BR"
    """Portuguese, Bralizian"""

    RO = "ro"
    """Romian"""

    FI = "fi"
    """Finnish"""

    SV_SE = "sv-SE"
    """Swedish, Sweden"""

    VI = "vi"
    """Vietnamese"""

    TR = "tr"
    """Turkish"""

    CS = "cs"
    """Czech"""

    EL = "el"
    """Greek"""

    BG = "bg"
    """Bulgarian"""

    RU = "ru"
    """Russian"""

    UK = "uk"
    """Ukrainian"""

    HI = "hi"
    """Hindi"""

    TH = "th"
    """Thai"""

    ZH_CN = "zh-CN"
    """Chinese, China"""

    JA = "ja"
    """Japanese"""

    ZH_TW = "zh-TW"
    """Chinese, Taiwan"""

    KO = "ko"
    """Korean"""
```

Comment (davfsa) [inline hikari/locales.py:127]: I dont think there is a need for the `Language is` prefix. Make it all a bit confusing
```suggestion
    DA = "da"
    """Danish"""

    DE = "de"
    """German"""

    EN_GB = "en-GB"
    """English, UK"""

    EN_US = "en-US"
    """English, US"""

    ES_ES = "es-ES"
    """Spanish, Spain"""

    FR = "fr"
    """French"""

    HR = "hr"
    """Croatian"""

    IT = "it"
    """Italian"""

    LT = "lt"
    """Lithuanian"""

    HU = "hu"
    """Hungarian"""

    NL = "nl"
    """Dutch"""

    NO = "no"
    """Norwegian"""

    OL = "pl"
    """Polish"""

    PT_BR = "pt-BR"
    """Portuguese, Bralizian"""

    RO = "ro"
    """Romian"""

    FI = "fi"
    """Finnish"""

    SV_SE = "sv-SE"
    """Swedish, Sweden"""

    VI = "vi"
    """Vietnamese"""

    TR = "tr"
    """Turkish"""

    CS = "cs"
    """Czech"""

    EL = "el"
    """Greek"""

    BG = "bg"
    """Bulgarian"""

    RU = "ru"
    """Russian"""

    UK = "uk"
    """Ukrainian"""

    HI = "hi"
    """Hindi"""

    TH = "th"
    """Thai"""

    ZH_CN = "zh-CN"
    """Chinese, China"""

    JA = "ja"
    """Japanese"""

    ZH_TW = "zh-TW"
    """Chinese, Taiwan"""

    KO = "ko"
    """Korean"""
```

File: hikari/locales.py [hunk 6]
```diff
@@ -37,91 +35,91 @@ class Locale(str, enums.Enum):
...
-    """Language is English, UK (en-GB)"""
+    """English, UK"""
 
     EN_US = "en-US"
-    """Language is English, US (en-US)"""
+    """English, US"""
 
     ES_ES = "es-ES"
-    """Language is Spanish (es-ES)"""
+    """Spanish"""
```
Comment (davfsa) [inline hikari/locales.py:50]:
```suggestion
    """Spanish, Spain"""
```

File: hikari/locales.py [hunk 7]
```diff
@@ -37,91 +35,91 @@ class Locale(str, enums.Enum):
...
-    """Language is Romian (ro)"""
+    """Romian"""
 
     FI = "fi"
-    """Language is Finnish (fi)"""
+    """Finnish"""
 
     SV_SE = "sv-SE"
-    """Language is Swedish (sv-SE)"""
+    """Swedish"""
```
Comment (davfsa) [inline hikari/locales.py:86]:
```suggestion
    """Swedish, Sweden"""
```

File: hikari/locales.py (hunk 6 above)
Comment (Le0Developer) [reply hikari/locales.py:50]: Name is directly from the Discord client, so I'm not sure if this should be changed.

Comment (davfsa) [reply hikari/locales.py:50]: The standard locale for those codes is the name I wanted to correct it to. But ig Discord just special like that, so nvm :)

**Review by davfsa — APPROVED** [review]

**Pull request merged**